### ISSUE TITLE
Question about “04_Support_Vector_Machines\04_Working_with_Kernels”

### ISSUE THREAD

**Issue by yingwei13mei** [nfmcclure/tensorflow_cookbook#26]
Hi, in 04_svm_kernels.py shows below, I have some questions to need your help, please.
1. In Line 54, the variable "model_output" seems not to be used in the whole py file, is there missed some codes?
2. Not very understand on how to calculate the loss function when use the my_kernel. From the code I guess the calculation is  that: my_kernel *(b^T *b) * (y * y^T) ,and then reduce_sum of it.
And the loss equals "reduce_sum(b)- the above result" 
So I don't understand why use this formula to get the loss function calculation result. Could you give me some help to understanding the process, please?
 "
\# Compute SVM Model 
54 model_output = tf.matmul(b, my_kernel) 
55 first_term = tf.reduce_sum(b) 
56 b_vec_cross = tf.matmul(tf.transpose(b), b) 
57 y_target_cross = tf.matmul(y_target, tf.transpose(y_target)) 
58 second_term = tf.reduce_sum(tf.multiply(my_kernel, tf.multiply(b_vec_cross, y_target_cross))) 
59 loss = tf.negative(tf.subtract(first_term, second_term)) 
"

**Comment by nfmcclure** [comment]
Hi @yingwei13mei ,  thanks for the notice.  You bring up a few points to address.

1. Yes, the 'model_output' is errant code that has been removed in the most recent pull request.
2. As for what exactly is going on in the SVM code, you might look into the following resources about SVMs:

http://cs229.stanford.edu/notes/cs229-notes3.pdf
http://www.robots.ox.ac.uk/~az/lectures/ml/lect3.pdf

3. I created a non-detailed jupyter notebook walkthrough of the code here:
https://github.com/nfmcclure/tensorflow_cookbook/blob/master/04_Support_Vector_Machines/04_Working_with_Kernels/04_svm_kernels.ipynb

Also, if you want to know what is going on in the RBF non-linear kernel calculations, I also created an jupyter walkthrough here:

https://github.com/nfmcclure/tensorflow_cookbook/blob/master/04_Support_Vector_Machines/05_Implementing_Nonlinear_SVMs/05_nonlinear_svm.ipynb

Good luck!
